app/src/main/java/org/wikipedia/richtext/CustomHtmlParser.kt

```diff
@@ -187,9 +189,36 @@ class CustomHtmlParser constructor(private val handler: TagHandler) : TagHandler
                         output.setSpan(TypefaceSpan("monospace"), start, output.length, 0)
                     }
                 }
+            } else if (tag == "ol") {
+                if (opening) {
+                    listParents.add(tag)
+                } else {
+                    listParents.remove(tag)
+                }
+                listItemCount = 0
+            } else if (tag == "li" && listParents.isNotEmpty() && !opening && output != null) {
+                handleListTag(output)
             }
             return false
         }
+
+        private fun handleListTag(output: Editable) {
+            if (listParents.last() == "ol") {
+                listItemCount++
+                val split = output.split("\n").filter { it.isNotEmpty() }
+                val start = output.length - split.last().length - 1
+                val replaceStr = "$listItemCount. ${split.last()}"
+                output.replace(start - 1, output.length - 1, replaceStr)
+
+                val spans = output.getSpans<LeadingMarginSpan>(output.length)
+                if (spans.isNotEmpty()) {
+                    val span = spans.last()
+                    val startSpan = output.getSpanStart(span)
+                    output.removeSpan(span)
+                    output.setSpan(LeadingMarginSpan.Standard(50 * listParents.size), startSpan, output.length, 0)
```

Comment:
What does the number `50` mean? If it's a dimension of pixels, it should probably be expressed in `dp` units, instead of raw pixels. (probably 16dp)

Reply:
Thanks, it makes sense to me. Done!